### ISSUE TITLE
Pug function problem!

### ISSUE THREAD

**Issue by jabaz** [yan-foto/electron-pug#6]
Following error comes when loading pug file:

> Error: the "basedir" option is required to use includes and extends with "absolute" paths

**Comment by jabaz** [comment]
I resolved this by adding basedir in options.

But there is also problem when in try catch, when error happens, `e.code is undefined`.
So adding this to 

```
if(typeof e.code === 'undefined' && typeof e.message === 'string') {
    return callback({data: new Buffer(`<pre style="tab-size:1">${e}</pre>`), mimeType:'text/html'});
}
```

before this

```
if (e.code.startsWith('PUG:')) {
    return callback({data: new Buffer(`<pre style="tab-size:1">${e}</pre>`), mimeType:'text/html'});
}
```

fixes problem!

**Comment by yan-foto** [comment]
Regarding the first comment, as you already have found the cause (see [here](https://github.com/pugjs/pug/blob/master/docs/views/api.pug#L36)), I must say that it has happened because of misconfiguration.

About the second comment, I need more information. What has caused the error? catching all exceptions is considered an anti pattern and I would avoid it!

**Comment by jabaz** [comment]
When i load pug file, and there is error in file, it crashes electron and causer is that e.code is undefined.
So in my case it happened, when trying to run function in pug, which is undefined.

**Comment by yan-foto** [comment]
All pug error _must_ start with `PUG:` (see `pug-error` package). If not, it is bug within pug package.

**Comment by jabaz** [comment]
I don't know why e.code is undefined! But i loaded file as it's descriped in this extension!

```
app.on('ready', () => {
  let mainWindow = new BrowserWindow({ width: 800, height: 600 });

  mainWindow.loadUrl(`file://${__dirname}/index.pug`);
  // the rest...
});
```

![image](https://cloud.githubusercontent.com/assets/13858505/15992933/39f9299a-30e1-11e6-9ee6-f9134adf0bbe.png)

It happens when trying to run function, which is undefined in pug!

```
doctype html
html
    head
        meta(charset='UTF-8')
    body
        header= test()
```

Here test is undefined!!!

And here is e printed out!!!

```
{ TypeError: D:/Git/Projects/other/hg-apps-electron/lib/public/index.pug:6
    4|         meta(charset='UTF-8')
    5|     body
  > 6|         header= test()

test is not a function
    at eval (eval at wrap (D:\Git\Projects\other\hg-apps-electron\node_modules\pug-runtime\wrap.js:6:10), <anonymous>:17:57)
    at template (eval at wrap (D:\Git\Projects\other\hg-apps-electron\node_modules\pug-runtime\wrap.js:6:10), <anonymous>:17:166)
    at Function.protocol.interceptBufferProtocol.e (D:\Git\Projects\other\hg-apps-electron\node_modules\electron-pug\index.js:64:59)
  path: 'D:/Git/Projects/other/hg-apps-electron/lib/public/index.pug' }
```

**Comment by yan-foto** [comment]
First things first: the catch block must be updated to handle errors with `undefined` codes.

Second: I am inclined to say that this behaviour should be regarded as a bug within pug. As `test` is considered to be function, pug tries to invoke it, just to find out that it doesn't exist.

One thing that bothers me is that `TypeError` should have only been thrown if `test` was already defined but wasn't a function (e.g. an `int`). In this case `test` isn't defined and the appropriate error should be `ReferenceError`. As I'm not aware of internal working of pug, I refrain to make any speculation as to why this happens!

I suggest to open an issue in pug repository and ask why errors caused by undefined functions does not have a code starting with `PUG`?

**Comment by jabaz** [comment]
Okay, thanks for the answer.
I may open issue to pug, when i got time!
Thanks!

**Comment by yan-foto** [comment]
Thank you for reporting. I will fix the catch block bug and will close this for now.

**Comment by jabaz** [comment]
They responded this, when i opened issued

> Error codes starting with PUG: are only used on errors emitted by Pug during compilation. In fact, the errors caught in the template itself are not passed through pug-error.

**Comment by yan-foto** [comment]
So, what would be the right behavior in your opinion?

I'm fine by just catching PUG and file not found errors and leave the rest up to the developer!

**Comment by yan-foto** [comment]
I fixed this in version `1.6.0`, now I don't check for `PUG:` prefix. It took two years to fix 😓
